src/planning.py

```diff
@@ -662,6 +717,110 @@ def _get_successor_with_correct_option(
     return action_seq
 
 
+def _sesame_plan_with_fast_downward(
+        task: Task, option_model: _OptionModelBase, nsrts: Set[NSRT],
+        initial_predicates: Set[Predicate], types: Set[Type], timeout: float,
+        seed: int, max_horizon: int,
+        optimal: bool) -> Tuple[List[_Option], Metrics]:
+    """A version of SeSamE that runs the Fast Downward planner to produce a
+    single skeleton, then calls run_low_level_search() to turn it into a plan.
+
+    Usage: Build and compile the Fast Downward planner, then set the environment
+    variable FD_EXEC_PATH to point to the `downward` directory. For example:
+    1) git clone https://github.com/ronuchit/downward.git
+    2) cd downward && ./build.py
+    3) export FD_EXEC_PATH="<your path here>/downward"
+    """
+    # Note: the types that would be extracted from the NSRTs here may not
+    # include all the environment's types, so it's better to use the
+    # types that are passed in as an argument instead.
+    nsrt_preds, _ = utils.extract_preds_and_types(nsrts)
+    # Ensure that initial predicates are always included.
+    predicates = initial_predicates | set(nsrt_preds.values())
+    init_atoms = utils.abstract(task.init, predicates)
+    objects = list(task.init)
+    start_time = time.time()
+    # Create the domain and problem strings, then write them to tempfiles.
+    dom_str = utils.create_pddl_domain(nsrts, predicates, types, "mydomain")
+    prob_str = utils.create_pddl_problem(objects, init_atoms, task.goal,
+                                         "mydomain", "myproblem")
+    dom_file = tempfile.NamedTemporaryFile(delete=False).name
+    with open(dom_file, "w", encoding="utf-8") as f:
+        f.write(dom_str)
+    prob_file = tempfile.NamedTemporaryFile(delete=False).name
+    with open(prob_file, "w", encoding="utf-8") as f:
+        f.write(prob_str)
+    sas_file = tempfile.NamedTemporaryFile(delete=False).name
```

Comment:
is the sas file used? or is the point just to not clutter the local dir?

Reply:
it's important that we give it a name, because otherwise FD defaults to using the name `sas.plan` or something like that, which will cause issues if you run multiple processes simultaneously (we've been burned by this before, more than once i think)

Reply:
add a comment to warn the future peeps